### PR TITLE
Strip server name's spaces from beginning and end

### LINKED ISSUE #402: Strip the beginning of a server name from repeating chars
**Is your feature request related to a problem? Please describe.**
People abuse this to get their server on top of the list when sorting alphabetically.

**Describe the solution you'd like**
- [ ] Remove leading spaces for all servers
- [ ] If a server contains no alphanumeric characters, hide from server list (this might be easier to enforce within the master server list?)
- [ ] When sorting by name it would be neat to only consider alphanumeric characters (CEGUI limitation?)

**Additional context**
https://bugs.mtasa.com/view.php?id=5070

### PR DESCRIPTION
**Pull request by Fernando-A-Rocha**
Closes #402

Trims leading and trailing spaces from the server name on config load.

Btw, on the topic of #402, it's not a good idea to add an error if server name doesn't contain an alphanumeric character because some foreign languages might not need them at all.

### CODE REVIEW

File: Server/mods/deathmatch/logic/CMainConfig.cpp [hunk 1]
```diff
@@ -133,6 +133,20 @@ bool CMainConfig::Load()
...
+    {
+        const size_t snameEnd = m_strServerName.find_last_not_of(" ");
+        m_strServerName = m_strServerName.substr(snameStart, snameEnd - snameStart + 1);
+    }
+    else
+    {
+        // The string contains only spaces
+        CLogger::ErrorPrintf("Server name must contain at least one character other than a space\n");
+        return false;
+    }
```
Comment (TracerDS) [inline Server/mods/deathmatch/logic/CMainConfig.cpp]: `std::size_t` instead of `size_t`.
Also maybe early return in case something would be changed here?

**Review by TracerDS — CHANGES_REQUESTED** [review]

File: Server/mods/deathmatch/logic/CMainConfig.cpp [hunk 2]
```diff
@@ -133,6 +133,20 @@ bool CMainConfig::Load()
         return false;
     }
 
+    // Strip spaces from beginning and end of server name
+    const size_t snameStart = m_strServerName.find_first_not_of(" ");
+    if (snameStart != std::string::npos)
+    {
+        const size_t snameEnd = m_strServerName.find_last_not_of(" ");
+        m_strServerName = m_strServerName.substr(snameStart, snameEnd - snameStart + 1);
+    }
```
Comment (TheNormalnij) [inline Server/mods/deathmatch/logic/CMainConfig.cpp]: It looks like a library function. You can add `trim` to `SString`

Comment (Fernando-A-Rocha) [reply Server/mods/deathmatch/logic/CMainConfig.cpp]: this is already the simplest and optimized we can do imo

Comment (TheNormalnij) [reply Server/mods/deathmatch/logic/CMainConfig.cpp]: String operations here are universal and could be reused in another code.
You may avoid out-of-context operations here and make the code more self-descriptive:
```c++
if (!m_strServerName.IsTrimed())
    m_strServerName = m_strServerName.Trim();
```

Comment (Fernando-A-Rocha) [reply Server/mods/deathmatch/logic/CMainConfig.cpp]: Done ty!

**Review by TracerDS — APPROVED** [review]

File: Server/mods/deathmatch/logic/CMainConfig.cpp [hunk 3]
```diff
@@ -133,6 +133,9 @@ bool CMainConfig::Load()
         return false;
     }
 
+    // Strip spaces from beginning and end of server name
+    m_strServerName = SString(m_strServerName).TrimStart(" ").TrimEnd(" ");
+
```
Comment (TheNormalnij) [inline Server/mods/deathmatch/logic/CMainConfig.cpp:138]: It's ok in this context, but TrimStart and TrimEnd suck

**Review by TheNormalnij — APPROVED** [review]

File: Server/mods/deathmatch/logic/CMainConfig.cpp (hunk 3 above)
Comment (Fernando-A-Rocha) [reply Server/mods/deathmatch/logic/CMainConfig.cpp:138]: Ok but why do they suck?

**Pull request merged**